pkg/scheduler/objects/events.go

```diff
@@ -50,6 +51,71 @@ func (evt *applicationEvents) sendPlaceholderLargerEvent(ph *Allocation, request
 	evt.eventSystem.AddEvent(event)
 }
 
+func (evt *applicationEvents) sendNewAllocationEvent(alloc *Allocation) {
+	if !evt.enabled {
+		return
+	}
+
+	event := events.CreateAppEventRecord(evt.app.ApplicationID, "", alloc.GetUUID(), si.EventRecord_ADD, si.EventRecord_APP_ALLOC, alloc.GetAllocatedResource())
+	evt.eventSystem.AddEvent(event)
+}
+
+func (evt *applicationEvents) sendNewAskEvent(request *AllocationAsk) {
+	if !evt.enabled {
+		return
+	}
+
+	event := events.CreateAppEventRecord(evt.app.ApplicationID, "", request.GetAllocationKey(), si.EventRecord_ADD, si.EventRecord_APP_REQUEST, request.GetAllocatedResource())
+	evt.eventSystem.AddEvent(event)
+}
+
+func (evt *applicationEvents) sendRemoveAllocationEvent(alloc *Allocation, terminationType si.TerminationType) {
+	if !evt.enabled {
+		return
+	}
+
+	var event *si.EventRecord
+	switch terminationType {
+	case si.TerminationType_UNKNOWN_TERMINATION_TYPE:
+		event = events.CreateAppEventRecord(evt.app.ApplicationID, "", alloc.GetUUID(), si.EventRecord_REMOVE, si.EventRecord_ALLOC_NODEREMOVED, alloc.GetAllocatedResource())
```

Comment:
This can be greatly simplified: everything is the same except the change detail. Just set a single variable based on `terminationType`.

Reply:
Updated it. Thanks.